rest-api/common-rest/src/main/java/uk/gov/gchq/gaffer/rest/factory/AbstractExamplesFactory.java

```diff
@@ -448,7 +448,7 @@ public ToMap toMap() {
 
     public GetWalks getWalks() {
         final List<String> edges = new ArrayList<>(getSchema().getEdgeGroups());
-        if (edges.isEmpty()) {
+        if (edges.isEmpty() || getSchema().getEntityGroups().isEmpty()) {
```

Comment:
So this is a lot more slimmer than the version we worked through, where we changed `getEntityId` and `getEdgeId`

Reply:
Seems to off/on with the cache graphs fine tho. so it works.

Reply:
The NPE was happening inside `getEntityId`, but every other usage of that function was checked, so I simply added a check here, rather than adding checks inside `getEntityId`.